javascript/ql/src/semmle/javascript/frameworks/PropertyProjection.qll

```diff
@@ -0,0 +1,151 @@
+/**
+ * Provides classes for modelling property projection functions.
+ *
+ * Subclass `PropertyProjection` to refine the behavior of the analysis on existing property projections.
+ * Subclass `CustomPropertyProjection` to introduce new kinds of property projections.
+ */
+
+import javascript
+
+/**
+ * A call that projects selected properties of an object.
```

Comment:
Here and below: these comments could do with examples of what is being modelled.